pkg/kv/kvserver/replica_raft.go

```diff
@@ -1584,6 +1614,9 @@ func (r *Replica) getAndGCSnapshotLogTruncationConstraintsLocked(
 		if recipientStore != 0 && item.recipientStore != recipientStore {
 			continue
 		}
+		if item.deadline != (time.Time{}) && ignoreDeadline {
```

Comment:
This seems named the wrong way around. If we ignore the deadline, shouldn't we look at all entries? We're ignoring entries that have a deadline, so consider renaming this param to `ignoreEntsWithDeadline`.
Also add a comment. It's not clear when the deadline is or isn't set. I think a deadline is set only before a transfer is inflight, but I am just guessing (and I wrote this code originally!)

Reply:
maybe `onlyInflight` is a better name.

Reply:
Thanks for taking a look! I've improved an existing comment and added another one explaining when the deadline is assigned. Let me know if you think it still needs to be improved.